### DISCUSSION TITLE
IP Address always reports as 127.0.0.1 in email logs

### DISCUSSION THREAD

**Discussion by evanferrao** [dani-garcia/vaultwarden#3485] [Q&A] [answered]
I've hosted vaultwarden on render.com using postgreSQL from bit.io and outlook for outgoing smtp.
I have setup a 'test' instance to verify the issue. 

https://vaultwardendeploy.onrender.com/

Steps to reproduce:
(1) Signup on above link
(2) Click mail confirmation link
(3) Login
(4) You should now get an email saying <Device/Browser> Logged in from IP 127.0.0.1 instead of the actual IP address :(
The same issue occurs for failed logins

Screenshot attached for reference
 
![Screenshot from 2023-04-29 04-12-44](https://user-images.githubusercontent.com/130160624/235265562-cfe609d6-085c-45a2-abe3-22f4d868d29f.png)
![Screenshot from 2023-04-29 04-14-41](https://user-images.githubusercontent.com/130160624/235266269-a6c2a193-fc2c-4edc-95cc-425a35bbbb51.png)

**Comment by BlackDex** [comment]
Set the correct `IP_HEADER`
https://github.com/dani-garcia/vaultwarden/blob/393645617efa6866001d40cc7f6a9485947e9b23/.env.template#L55...L57

**Reply by evanferrao** [reply]
> Set the correct `IP_HEADER` https://github.com/dani-garcia/vaultwarden/blob/393645617efa6866001d40cc7f6a9485947e9b23/.env.template#L55...L57

I've tried changing it to `none` but to no avail. I also tried `X-Real-IP` (which is supposedly the default), but the result is the same. The email logs always report `127.0.0.1`
Is there, perhaps, any other option beside these two?

I've attached the screenshot of the config below
![Screenshot from 2023-04-29 15-38-11](https://user-images.githubusercontent.com/130160624/235297359-2804b51f-135e-436c-9eca-e528a5a6e707.png)

**Reply by BlackDex** [reply]
Depending on what your reverse proxy sends by either default, or whatever you configured you need to use that.

`none` will not work, as that is probably not a header sent by the reverse proxy.

Check https://github.com/dani-garcia/vaultwarden/wiki/Proxy-examples for more details on the configuration of the reverse proxies.

**Reply by evanferrao** [reply]
In that case I don't think this would be possible, as render has multiple outbound IP's and they change when the service reboots
Thanks for taking the time to reply, anyway :)

![image](https://user-images.githubusercontent.com/130160624/235299142-c57f944b-81a9-4738-b602-2cdbf5cefcf4.png)

**Reply by BlackDex** [reply, marked as answer]
I don't understand what outbound ip's have to do with this. It's about inbound. The proxy you use probably already send a header with the clients ip. Check the admin interface and go to diagnostics it tries to detect some common headers.

**Reply by evanferrao** [reply]
I'm sorry, I didn't get it the first time. I now understood it. Furthermore, I checked in the admin panel and it was ` X-Forwarded-For`. The real IP Address now shows in email reports/logs.
Thank you once again.

**Comment by evanferrao** [comment]
How do I mark a discussion as closed by "selecting an answer"?

**Reply by BlackDex** [reply]
No clue. Seems the mark as answer button is not visible for some reason.

**Reply by stefan0xC** [reply]
I think that it depends on the [category](https://docs.github.com/en/discussions/managing-discussions-for-your-community/managing-categories-for-discussions). I.e. this feature is only available in [Q&A](https://github.com/dani-garcia/vaultwarden/discussions/categories/q-a)

**Reply by evanferrao** [reply]
Yeah, it worked. Thanks.
